tests/umock_lock_factory_pthread_ut/umock_lock_factory_pthread_ut.c

```diff
@@ -0,0 +1,511 @@
+    if (new_calls != NULL)
+    {
+        mocked_calls = new_calls;
+        mocked_calls[mocked_call_count].call_type = TEST_MOCK_CALL_TYPE_mock_pthread_rwlock_init;
+        mocked_calls[mocked_call_count].u.mock_pthread_rwlock_init.rwlock = rwlock;
+        mocked_calls[mocked_call_count].u.mock_pthread_rwlock_init.attr = attr;
+        mocked_call_count++;
+    }
+    return mock_pthread_rwlock_init_result;
+}
+
+int mock_pthread_rwlock_rdlock(pthread_rwlock_t *rwlock)
+{
+    TEST_MOCK_CALL* new_calls = (TEST_MOCK_CALL*)realloc(mocked_calls, sizeof(TEST_MOCK_CALL) * (mocked_call_count + 1));
+    if (new_calls != NULL)
+    {
+        mocked_calls = new_calls;
+        mocked_calls[mocked_call_count].call_type = TEST_MOCK_CALL_TYPE_mock_pthread_rwlock_rdlock;
+        mocked_calls[mocked_call_count].u.mock_pthread_rwlock_rdlock.rwlock = rwlock;
+        mocked_call_count++;
+    }
+}
+
+int mock_pthread_rwlock_unlock(pthread_rwlock_t *rwlock)
+{
+    TEST_MOCK_CALL* new_calls = (TEST_MOCK_CALL*)realloc(mocked_calls, sizeof(TEST_MOCK_CALL) * (mocked_call_count + 1));
+    if (new_calls != NULL)
+    {
+        mocked_calls = new_calls;
+        mocked_calls[mocked_call_count].call_type = TEST_MOCK_CALL_TYPE_mock_pthread_rwlock_unlock;
+        mocked_calls[mocked_call_count].u.mock_pthread_rwlock_unlock.rwlock = rwlock;
+        mocked_call_count++;
+    }
+}
+
+int mock_pthread_rwlock_wrlock(pthread_rwlock_t *rwlock)
+{
+    TEST_MOCK_CALL* new_calls = (TEST_MOCK_CALL*)realloc(mocked_calls, sizeof(TEST_MOCK_CALL) * (mocked_call_count + 1));
+    if (new_calls != NULL)
+    {
+        mocked_calls = new_calls;
+        mocked_calls[mocked_call_count].call_type = TEST_MOCK_CALL_TYPE_mock_pthread_rwlock_wrlock;
+        mocked_calls[mocked_call_count].u.mock_pthread_rwlock_wrlock.rwlock = rwlock;
+        mocked_call_count++;
+    }
+}
+
+int mock_pthread_rwlock_destroy(pthread_rwlock_t *rwlock)
+{
+    TEST_MOCK_CALL* new_calls = (TEST_MOCK_CALL*)realloc(mocked_calls, sizeof(TEST_MOCK_CALL) * (mocked_call_count + 1));
+    if (new_calls != NULL)
+    {
+        mocked_calls = new_calls;
+        mocked_calls[mocked_call_count].call_type = TEST_MOCK_CALL_TYPE_mock_pthread_rwlock_destroy;
+        mocked_calls[mocked_call_count].u.mock_pthread_rwlock_destroy.rwlock = rwlock;
+        mocked_call_count++;
+    }
+}
+
+static void reset_all_calls(void)
+{
+    malloc_call_count = 0;
+    when_shall_malloc_fail = 0;
+
+    mock_pthread_rwlock_init_result = 0;
+    mock_pthread_rwlock_rdlock_result = 0;
+    mock_pthread_rwlock_unlock_result = 0;
+    mock_pthread_rwlock_wrlock_result = 0;
+    mock_pthread_rwlock_destroy_result = 0;
+
+    if (mocked_calls != NULL)
+    {
+        free(mocked_calls);
+        mocked_calls = NULL;
+    }
+
+    mocked_call_count = 0;
+}
+
+static TEST_MUTEX_HANDLE test_mutex;
+static TEST_MUTEX_HANDLE global_mutex;
+
+BEGIN_TEST_SUITE(umock_lock_factory_windows_unittests)
```

Comment:
>_windows [](start = 35, length = 8)

pthread

Reply:
Fixed.

---
In reply to: [530660377](https://github.com/Azure/umock-c/pull/177#discussion_r530660377) [](ancestors = 530660377)